### ISSUE TITLE
Update/ Create Listing via OKAPI

### ISSUE THREAD

**Issue by protux** [opencaching/okapi#440]
Hi,

for some apps it would be useful to update a cache-listing via okapi. e.g. one problem are double listings which are published on OC and GC but will only be maintained on GC. An app could automatically offer to keep an eye on the GC listing and update the OC listing.

Without the api one would need to store the users credentials what would not be good. So this "quality assurance" tool could only exist with the help of the OKAPI.

Another use case could be if a user is fed up with GC and whants to move his account to OC (or maybe he decides to create double listings of all his caches). If he wants to move his caches he would need to do that manually, with OKAPI support one could automate that.

What do you think about listing manipulation via OKAPI?

**Comment by wrygiel** [comment]
Adding new log entries is already implemented. So, what you propose is to add methods for editing past log entries, right?

**Comment by wrygiel** [comment]
I feel there are too many issues in one issue here. :) I got confused a bit.

**Comment by protux** [comment]
No, my proposal aims more for editing the cache description incl. D/T Value, coordinates, Spoilerimages etc. :) Maybe listing was the wrong word, excuse me, please.

**Comment by wrygiel** [comment]
I'm afraid this won't be possible at this point. There are too many differences between both OC branches to allow us to create one consistent API for that.

And even if this changed, it would still be a huge task and, sadly, we probably won't have resources to ever do this. (I even think there already was an issue for that.)

**Comment by following5** [comment]
#488 is a subissue of this, adding a method to change geocache listings via OKAPI. It starts with changing passwords, which is rather simple.

Expanding that to editing or submitting whole geocache listings will be very expensive, as Wrygiel wrote. But I think it's possible, the (relative) differences between branches are similar to submitting logs.

And this feature is really needed, exactly for the reasons that protux gave. For OCDE there has been a tool that copies GC listings to OC by directly interacting with the OC.de website, but it has recently been broken by OCDE code changes and probably won't come back. OKAPI would be the logical way to implement a replacement.

**Comment by following5** [comment]
Editable properties of a Geocache:
* name (only one name on OC sites, names in multiple languages in OKAPI)
* type
* container size
* coordinates
* country
* OCPL: region (automatically calculated in OCDE)
* difficulty
* terrain
* trip time
* trip distance
* GC code
* OCPL: NC code
* OCPL: TC code
* OCPL: GPSGames code
* attributes
* descriptions in multiple languages
* short descriptions in multiple languages
* hints in multiple languages
* images
  * title
  * binary image data
  * position
  * spoiler flag
  * don't display as attachement flag
  * OCDE: map preview flag
* additional waypoints
  * type
  * coordinates
  * plaintext description
  * OCPL: stage number
  * OCPL: display option
* OCPL: MP3s
  * title
  * binary mp3 data
  * don't display as attachement flag
* hide date / event date
* password for 'found it' and 'attended' logs (IMPLEMENTED)
* OCDE: cache lists display flag
* OCDE: protect old coords flag

**Comment by following5** [comment]
The interface of `services/caches/add` may be identical to `services/caches/edit`, with these exceptions:
* 'cache_code' is a return value instead of a parameter when adding.
* At least 10 of the 27 fields are mandantory for adding (the first 9 and the hide/event date), while all are optional for editing.
* different allowed values for 'status'

**Comment by following5** [comment]
Implementation of full geocache edting is blocked by several issues, see https://github.com/opencaching/okapi/pull/570#issuecomment-455196438.

Implementation of submitting caches is additionally blocked by #565.
